### PR TITLE
Add Useful Resources and Example Build Instructions to CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by HudsonHumphries**
These are resources I found helpful as I was getting to learn about Open Telemetry and specifically the C++ repository. Adding these resources will help newcomers as they will have a central place to go to for resources. This Open Telemetry Onboarding Resources Guide was written by @hudsonhumphries, @ankit-bhargava and @Brandon-Kimberly

### CODE REVIEW

**Comment by reyang** [comment]
> I wonder about merging this with the [CONTRIBUTING.md](https://github.com/open-telemetry/opentelemetry-cpp/blob/master/CONTRIBUTING.md).

+1

**Comment by HudsonHumphries** [comment]
Thanks for the feedback! I am working with Ankit who made [PR 94](https://github.com/open-telemetry/opentelemetry-cpp/pull/94) to move our documentation into CONTRIBUTING.md.

File: docs/open-telemetry-onboarding-resource-guide.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+* Read through the [Open Telemetry Collector](https://github.com/open-telemetry/opentelemetry-collector) GitHub repository
+    * This repository has a lot of good information surrounding the Open Telemetry ecosystem. At the top of the **[readme](https://github.com/open-telemetry/opentelemetry-collector/blob/master/README.md)**, there are multiple links that give newcomers a good idea of what the project is about and how to get involved in it.
+* Read through the Open Telemetry Python documentation for its [API](https://opentelemetry-python.readthedocs.io/en/stable/api/api.html) and [SDK](https://opentelemetry-python.readthedocs.io/en/stable/sdk/sdk.html)
+    * The [API](https://github.com/open-telemetry/opentelemetry-java/blob/master/QUICKSTART.md) and [SDK](https://github.com/open-telemetry/opentelemetry-java/blob/master/QUICKSTART.md) documentation provides a lot of information on what the classes and their functions are used for. Since there is currently minimal documentation for C++, use the Python repository’s extensive documentation to learn more about how the API and SDK work.
```
Comment (g-easy) [inline docs/open-telemetry-onboarding-resource-guide.md:18]: Link is to Java but no mention of Java in the text?

Comment (HudsonHumphries) [reply docs/open-telemetry-onboarding-resource-guide.md:18]: I believe those should be linking to the API and SDK documentation for Python, thanks for catching that. Will fix it

**Comment by HudsonHumphries** [comment]
As Ankit mentioned in his PR, we moved build instructions for examples into this PR. I also moved the resources into CONTRIBUTING.md and fixed bad links.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -72,3 +72,60 @@ A PR is considered to be **ready to merge** when:
...
+### Code Examples
+
+* Follow the [simple trace example](https://github.com/open-telemetry/opentelemetry-cpp/pull/92) for an introduction to basic Open Telemetry functionality in C++.  Currently the example can be found in [PR #94](https://github.com/open-telemetry/opentelemetry-cpp/pull/94)
+
+* Read through the [Java Quick-Start Guide](https://github.com/open-telemetry/opentelemetry-java/blob/master/QUICKSTART.md)
+    * This shows you how the classes and functions will interact in simple and easy to digest examples.
+* Take a look at this [Java SDK example.](https://github.com/open-telemetry/opentelemetry-java/tree/master/examples/sdk-usage) This shows a good use case of the SDK using stdout exporter.
+* Take a look at the [Java Jaeger example.](https://github.com/open-telemetry/opentelemetry-java/tree/master/examples/jaeger) This provides a brief introduction to the Jaeger exporter, its interface, and how to interact with the service.
+
+Please contribute! You’re welcome to add more information if you come across any helpful resources
```
Comment (reyang) [inline CONTRIBUTING.md:131]: nit: missing an end period.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -72,3 +72,60 @@ A PR is considered to be **ready to merge** when:
 * Urgent fixes can take exceptions as long as it has been actively communicated.
 
 Any Approver / Maintainer can merge the PR once it is **ready to merge**.
+
+## Build and Run Code Examples
```
Comment (reyang) [inline CONTRIBUTING.md:76]: I would suggest putting this section under `Development` (above `How to Send Pull Requests`).

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -72,3 +99,33 @@ A PR is considered to be **ready to merge** when:
...
+    * This repository has a lot of good information surrounding the Open Telemetry ecosystem. At the top of the **[readme](https://github.com/open-telemetry/opentelemetry-collector/blob/master/README.md)**, there are multiple links that give newcomers a good idea of what the project is about and how to get involved in it.
+* Read through the Open Telemetry Python documentation
+    * The [API](https://opentelemetry-python.readthedocs.io/en/stable/api/api.html) and [SDK](https://opentelemetry-python.readthedocs.io/en/stable/sdk/sdk.html) documentation provides a lot of information on what the classes and their functions are used for. Since there is currently minimal documentation for C++, use the Python repository’s extensive documentation to learn more about how the API and SDK work.
+
+### Code Examples
+
+* Follow the [simple trace example](https://github.com/open-telemetry/opentelemetry-cpp/pull/92) for an introduction to basic Open Telemetry functionality in C++.  Currently the example can be found in [PR #94](https://github.com/open-telemetry/opentelemetry-cpp/pull/94)
```
Comment (pyohannes) [inline CONTRIBUTING.md:124]: When this is merged, can you submit an issue that reminds us of updating this document before we do a first release? Once we have our own examples and exporters in place, many of the examples here should be switched out.

Comment (HudsonHumphries) [reply CONTRIBUTING.md:124]: Yes, I'll get that issue up once this PR this merged

**Comment by reyang** [comment]
@pyohannes mentioned that he is okay to merge this PR during the community meeting.
Please rebase, thanks.

**Comment by HudsonHumphries** [comment]
Rebased @reyang

**Pull request merged**